src/cocotb/_scheduler.py

```diff
@@ -172,6 +173,12 @@ def thread_wait(self):
         return self.state
 
 
+class GPIState(Enum):
```

Comment:
Will this be the only state of the GPI "forever", or should this be a more specific name in case of other states?

Reply:
This is specifically for tracking GPI states so we can do things like preventing registering ReadWrite in ReadOnly stage. If there end up being different kinds of states, they will be in a different variable. I can take an action to change the variable name to `_gpi_state`.